.travis.yml

```diff
@@ -7,15 +7,39 @@ sudo: required
 before_install:
   - if [[ "$INSTALL_ARM_DEPS" == "yes" ]]; then tools/apt-get-install-arm.sh; fi
   - if [[ "$INSTALL_NUTTX_DEPS" == "yes" ]]; then tools/apt-get-install-nuttx.sh; fi
+  - if [[ "$INSTALL_TRAVIS_I686_DEPS" == "yes" ]]; then tools/apt-get-install-travis-i686.sh; fi
   - tools/apt-get-install-deps.sh
 
 install:
 
 script: "tools/precommit.py $OPTS"
 
 env:
-  - OPTS="--test=host --buildtype=release"
-  - OPTS="--test=host --buildtype=debug"
+  - OPTS="--test=host"
   - OPTS="--test=rpi2" INSTALL_ARM_DEPS=yes
   - OPTS="--test=nuttx" INSTALL_NUTTX_DEPS=yes
   - OPTS="--test=misc"
+
+matrix:
```

Comment:
My take is that this change brings us towards being able to detect build breaks on a single x86_64 machine.
So far so good.

Still, AFAIU the matrix feature is broken on travis (https://github.com/travis-ci/travis-ci/issues/1519), so why pretend we do it here? precommit.py handles some of the option setup, and this change provides another source of compiler flags, cmake defines and such.
Why can't we have it defined what is built/tested in a single place?